dvclive/metrics.py

```diff
@@ -49,23 +49,44 @@ def __init__(
                 ) from exception
 
     @staticmethod
-    def from_env():
+    def from_env(existing_logger: "MetricLogger" = None):
         from . import env
 
         if env.DVCLIVE_PATH in os.environ:
             directory = os.environ[env.DVCLIVE_PATH]
-            dump_latest = bool(int(os.environ.get(env.DVCLIVE_SUMMARY, "0")))
-            html = bool(int(os.environ.get(env.DVCLIVE_HTML, "0")))
-            checkpoint = bool(int(os.environ.get(env.DVC_CHECKPOINT, "0")))
-            resume = bool(int(os.environ.get(env.DVCLIVE_RESUME, "0")))
-            return MetricLogger(
-                directory,
-                summary=dump_latest,
-                html=html,
-                checkpoint=checkpoint,
-                resume=resume,
-            )
-        return None
+            env_config = {
+                "summary": bool(int(os.environ.get(env.DVCLIVE_SUMMARY, "0"))),
+                "html": bool(int(os.environ.get(env.DVCLIVE_HTML, "0"))),
+                "checkpoint": bool(
+                    int(os.environ.get(env.DVC_CHECKPOINT, "0"))
+                ),
+                "resume": bool(int(os.environ.get(env.DVCLIVE_RESUME, "0"))),
+            }
+
+            if existing_logger:
+
+                def config_matches(metric_logger: MetricLogger):
+                    return (
+                        env_config["summary"] == metric_logger._summary
+                        and env_config["html"] == metric_logger._html
+                        and env_config["checkpoint"]
+                        == metric_logger._checkpoint
+                    )
+
+                if existing_logger.dir == directory and config_matches(
+                    existing_logger
+                ):
+                    return existing_logger
+                else:
+                    logger.info(
+                        "Dvclive logger config ('%s') has changed. "
+                        "New logger will write to '%s'.",
+                        existing_logger.dir,
+                        directory,
+                    )
+
+            return MetricLogger(directory, **env_config)
```

Comment:
OK so if the env vars don't match the existing logger, a new logger is created... Seems correct i.e. this does what the PR states. But I'm not sure I get how this addresses #60. Need to think... ⌛

Reply:
My q was answered in https://github.com/iterative/dvclive/issues/60#issuecomment-843941753:

> when dvc provides --live option for stage/run, it is equivalent to setting the environmental variables that dvclive can understand ... existence of those env vars means that we are running in the context of dvc

It's how DVC and Dvclive communicate. Got it 👍